khelo/lib/ui/flow/settings/support/contact_support_view_model.dart

```diff
@@ -0,0 +1,151 @@
+      ref.read(currentUserPod)?.id);
+});
+
+class ContactSupportViewStateNotifier
+    extends StateNotifier<ContactSupportViewState> {
+  final ImagePicker picker;
+  final FileUploadService fileUploadService;
+  final SupportService supportService;
+  final String? _currantUserId;
+
+  ContactSupportViewStateNotifier(
+    this.picker,
+    this.fileUploadService,
+    this.supportService,
+    this._currantUserId,
+  ) : super(ContactSupportViewState(
+          titleController: TextEditingController(),
+          descriptionController: TextEditingController(),
+        ));
+
+  void onValueChange() {
+    state = state.copyWith(
+      actionError: null,
+      enableSubmit: state.titleController.text.trim().isNotEmpty &&
+          state.attachments
+                  .where((element) => element.uploadStatus.isUploading)
+                  .firstOrNull ==
+              null,
+    );
+  }
+
+  void _uploadAttachments({required String path, required String name}) async {
+    try {
+      state = state.copyWith(attachments: [
+        ...state.attachments,
+        Attachment(
+            path: path,
+            name: name,
+            uploadStatus: AttachmentUploadStatus.uploading),
+      ]);
+      onValueChange();
+      final bool isLarge = await File(path).length() > 25 * 1024 * 1024;
+      if (isLarge) {
+        final attachments = state.attachments.toList();
+        attachments.removeWhere((element) => element.path == path);
+        state = state.copyWith(
+            attachments: attachments,
+            actionError:
+                "Oops! Your file exceeds the maximum allowed size of 25 MB. Please choose a smaller file and try again.");
+      } else {
+        final url = await fileUploadService.uploadProfileImage(
+            path, ImageUploadType.support);
+        state = state.copyWith(
+          actionError: null,
+          attachments: state.attachments.updateWhere(
+            where: (attachment) => attachment.path == path,
+            updated: (oldElement) => oldElement.copyWith(
+              url: url,
+              uploadStatus: AttachmentUploadStatus.uploaded,
+            ),
+          ),
+        );
+      }
+
+      onValueChange();
+    } catch (error, _) {
+      state = state.copyWith(actionError: error);
+      debugPrint(
+          "ContactSupportViewStateNotifier: Error while uploading $path error $error");
+    }
+  }
+
+  void pickAttachments() async {
+    try {
+      final List<XFile> medias = await picker.pickMultipleMedia(
+        imageQuality: 70,
+      );
+      for (XFile media in medias) {
+        _uploadAttachments(path: media.path, name: media.name);
+      }
+    } catch (error, _) {
+      debugPrint(
+        "ContactSupportViewStateNotifier: Error while pick image! $error",
+      );
+    }
+  }
+
+  void removeAttachment(int index) {
+    state = state.copyWith(
+      attachments: state.attachments.toList()..removeAt(index),
+    );
+    onValueChange();
+  }
```

Comment:
we are removing attachments locally only, shouldn't we remove it from database as well if user is not going to use that image while adding support cases.